trunk/SUAVE/Methods/Aerodynamics/Common/Fidelity_Zero/Lift/weissinger_vortex_lattice.py

```diff
@@ -80,18 +82,88 @@ def weissinger_vortex_lattice(conditions,configuration,wing):
 
     if orientation == False :
 
-        # discretizing the wing sections into panels            
-        i              = np.arange(0,n)
-        section_length = dchord/span*(span-(i+1)*deltax+deltax/2) + tip_chord
-        twist_distri   = twist_rc + i/float(n)*(twist_tc-twist_rc)
-        
-        ya = np.atleast_2d((i)*deltax)
-        yb = np.atleast_2d((i+1)*deltax)
-        xa = np.atleast_2d(((i+1)*deltax-deltax/2)*np.tan(sweep) + 0.25*section_length)
-        x  = np.atleast_2d(((i+1)*deltax-deltax/2)*np.tan(sweep) + 0.75*section_length)
-        y  = np.atleast_2d(((i+1)*deltax-deltax/2))      
+        # Determine if wing segments are defined  
+        segment_keys = wing.Segments.keys()
+        n_segments   = len(segment_keys)
+        segment_vortex_index = np.zeros(n_segments)
+        # If spanwise stations are setup
+        if n_segments>0:
+            # discretizing the wing sections into panels
+            i              = np.arange(0,n)
+            j              = np.arange(0,n+1)
+            y_coordinates = (j)*deltax             
+            segment_chord = np.zeros(n_segments)
+            segment_twist = np.zeros(n_segments)
+            segment_sweep = np.zeros(n_segments)
+            segment_span = np.zeros(n_segments)
+            segment_chord_x_offset = np.zeros(n_segments)
+            section_stations = np.zeros(n_segments)
+            
+            # obtain chord and twist at the beginning/end of each segment
+            for i_seg in xrange(n_segments):                
+                segment_chord[i_seg] = wing.Segments[segment_keys[i_seg]].root_chord_percent*root_chord
+                segment_twist[i_seg] = wing.Segments[segment_keys[i_seg]].twist
+                segment_sweep[i_seg] = wing.Segments[segment_keys[i_seg]].sweeps.quarter_chord
+                section_stations[i_seg] = wing.Segments[segment_keys[i_seg]].percent_span_location*span
+                
+                if i_seg == 0:
+                    segment_span[i_seg] = 0.0
+                    segment_chord_x_offset[i_seg] = 0.25*root_chord # weissinger uses quarter chord as reference
+                else:
+                    segment_span[i_seg]    = wing.Segments[segment_keys[i_seg]].percent_span_location*span - wing.Segments[segment_keys[i_seg-1]].percent_span_location*span
+                    segment_chord_x_offset[i_seg]  = segment_chord_x_offset[i_seg-1] + segment_span[i_seg]*np.tan(segment_sweep[i_seg-1])
+            
+            # shift spanwise vortices onto section breaks 
+            for i_seg in xrange(n_segments):
```

Comment:
there's probably a way to do this without a for loop

Reply:
Requires Loop. Issue resolved